gateway/utils/syncman/store_sc.go

```diff
@@ -0,0 +1,195 @@
+package syncman
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/spaceuptech/space-cloud/gateway/config"
+)
+
+type KubeStore struct {
+	clusterID string
+	kube      *kubernetes.Clientset
+}
+
+type scTableScheam struct {
+	clusterId string          `json:"clusterId"`
+	ProjectId string          `json:"projectId"`
+	Project   *config.Project `json:"project"`
+}
+
+func NewKubeStore(clusterID string) (*KubeStore, error) {
+	// Create the kubernetes client
+	restConfig, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	kube, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &KubeStore{clusterID: clusterID, kube: kube}, nil
+}
+
+func (s *KubeStore) Register() {
+	// kubernetes will handle this automatically
+}
+
+func (s *KubeStore) WatchProjects(cb func(projects []*config.Project)) error {
+	labels := fmt.Sprintf("app=%s,clusterId=%s", "project", s.clusterID) // todo verify this
+	watcher, err := s.kube.CoreV1().ConfigMaps(fmt.Sprintf("sc/projects/%s", s.clusterID)).Watch(v12.ListOptions{Watch: true, LabelSelector: labels})
+	if err != nil {
+		return err
+	}
+	defer watcher.Stop()
+
+	projectMap := map[string]scTableScheam{}
+	for ee := range watcher.ResultChan() {
+		switch ee.Type {
+		case watch.Added, watch.Modified:
+			configMap := ee.Object.(*v1.ConfigMap)
+			for key, value := range configMap.Data {
```

Comment:
Data will have only one key namely `project` which contains the Project information in json. There will be another key `id` which holds the projectid

Reply:
then how does kubernetes differenciates between different projects in same cluster

Reply:
labels?